File: back-end/patient-care/src/main/java/com/hims/patient_care/service/DoctorVisitService.java
package com.hims.patient_care.service;


import java.util.List;



import com.hims.patient_care.model.DoctorVisit;

public interface DoctorVisitService {


	List<DoctorVisit> getVisitsByDoctorId(Long doctorId);

	DoctorVisit createVisit(DoctorVisit visit);

}

File: front-end/healthcare-dashboard/src/app/services/doctor-visit.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { DoctorVisit } from '../classes/doctorvisit.model';

@Injectable({
  providedIn: 'root'
})
export class DoctorVisitService {

  private apiUrl = 'http://localhost:8080/api/doctor-visits'; // Spring Boot API URL

  constructor(private http: HttpClient) { }

  getVisits(doctorId: number): Observable<DoctorVisit[]> {
    return this.http.get<DoctorVisit[]>(`${this.apiUrl}/doctor/${doctorId}`);
  }

  createVisit(visit: DoctorVisit): Observable<DoctorVisit> {
    return this.http.post<DoctorVisit>(`${this.apiUrl}`, visit);
  }
}
